contracts/votingMachines/GenesisProtocolLogic.sol

```diff
@@ -0,0 +1,660 @@
+        require(_params[8] > 0,"minimumDaoBounty should be > 0");
+        require(_params[9] > 0,"daoBountyConst should be > 0");
+
+        bytes32 paramsHash = getParametersHash(_params, _voteOnBehalf);
+
+        int alpha = int216(_params[4]).fraction(int216(1000));
+        //set a limit for power for a given alpha to prevent overflow
+        uint limitExponent = 172;//for alpha less or equal 2
+        uint j = 2;
+        for (uint i = 2;i < 16 ;i = i*2) {
+            if ((uint(alpha.fromReal()) > i) && (uint(alpha.fromReal()) <= i*2)) {
+                limitExponent = limitExponent/j;
+                break;
+             }
+            j++;
+        }
+
+        parameters[paramsHash] = Parameters({
+            queuedVoteRequiredPercentage: _params[0],
+            queuedVotePeriodLimit: _params[1],
+            boostedVotePeriodLimit: _params[2],
+            preBoostedVotePeriodLimit: _params[3],
+            thresholdConstA:alpha,
+            limitExponentValue:limitExponent,
+            quietEndingPeriod: _params[5],
+            proposingRepReward: _params[6],
+            votersReputationLossRatio:_params[7],
+            minimumDaoBounty:_params[8],
+            daoBountyConst:_params[9],
+            activationTime:_params[10],
+            voteOnBehalf:_voteOnBehalf
+        });
+        return paramsHash;
+    }
+
+  /**
+   * @dev hashParameters returns a hash of the given parameters
+   */
+    function getParametersHash(
+        uint[11] _params,//use array here due to stack too deep issue.
+        address _voteOnBehalf
+    )
+        public
+        pure
+        returns(bytes32)
+        {
+        //double call to keccak256 to avoid deep stack issue when call with too many params.
+        return keccak256(
+            abi.encodePacked(
+             keccak256(
+              abi.encodePacked(
+                _params[0],
+                _params[1],
+                _params[2],
+                _params[3],
+                _params[4],
+                _params[5],
+                _params[6],
+                _params[7],
+                _params[8],
+                _params[9],
+                _params[10]
+             )),
+            _voteOnBehalf
+        ));
+    }
+
+    /**
+      * @dev execute check if the proposal has been decided, and if so, execute the proposal
+      * @param _proposalId the id of the proposal
+      * @return bool true - the proposal has been executed
+      *              false - otherwise.
+     */
+    function _execute(bytes32 _proposalId) internal votable(_proposalId) returns(bool) {
+        Proposal storage proposal = proposals[_proposalId];
+        Parameters memory params = parameters[proposal.paramsHash];
+        Proposal memory tmpProposal = proposal;
+        uint totalReputation = VotingMachineCallbacksInterface(proposal.callbacks).getTotalReputationSupply(_proposalId);
+        uint executionBar = totalReputation * params.queuedVoteRequiredPercentage/100;
```

Comment:
rename to "votingBar"

Reply:
why ?

Reply:
the proposal can be executed regardless of this bar.  the bar relates directly to voting, less so to execution.

Reply:
how about absoluteMajorityBar ?

Reply:
perfect